### PR TITLE
Move KubeRay autoscaler files into Ray autoscaler directory, add an entry-point.

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

This PR consists of the following clean-up items for KubeRay autoscaler integration:
- Remove the `docker/kuberay` directory
  * Move the Python files formerly in `docker/kuberay` to the autoscaler directory.
  * Use a `rayproject/ray` image for the autoscaler.
- Add an entry point for the kuberay autoscaler to `scripts.py`. Use the entry point in the example config.
- Slightly simplify the code that starts the autoscaler.
- Ray versions are updated to Ray 1.11.0, which will be officially released within the next couple of days.
- By default, Ray >= 1.11.0 runs without Redis. References to Redis are removed from the example config. 
- Add the autoscaler configuration test to the CI.
- Update development documentation to reflect the changes in this PR.

Another thing to note:

The image used for the autoscaler is now `rayproject/ray:nightly`.
After this is merged, we will have to wait an hour or so while a nightly image with these changes is built.
The image can then be to a particular Ray commit in a follow-up PR.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(
Tested manually, CI tests coming soon.

### CODE REVIEW

**Comment by DmitriGekhtman** [comment]
cc @Jeffwan

File: python/ray/scripts/scripts.py [hunk 1]
```diff
@@ -2003,6 +2004,37 @@ def global_gc(address):
...
+    "Should coincide with the `metadata.name` of the RayCluster CR.",
+)
+@click.option(
+    "--cluster-namespace",
+    required=True,
+    type=str,
+    help="The Kubernetes namespace the Ray Cluster lives in.\n"
+    "Should coincide with the `metadata.namespace` of the RayCluster CR.",
+)
+@click.option(
```
Comment (wuisawesome) [inline python/ray/scripts/scripts.py:2022]: remove option?

**Review by wuisawesome — APPROVED** [review]
looks like one remaining reference to redis passwords, but other than that lgtm

File: python/ray/scripts/scripts.py (hunk 1 above)
Comment (DmitriGekhtman) [reply python/ray/scripts/scripts.py:2022]: From what I understand, we may want to leave the possibility of using Redis to users who require it.
This is basically a private API and the Redis password is and optional option.

Comment (DmitriGekhtman) [reply python/ray/scripts/scripts.py:2022]: Let me change the default to the empty string "", though for consistency with the rest of the call sites.

Also, let me add a warning to the help for this option.

**Comment by Jeffwan** [comment]
> Ray versions are updated to Ray 1.11.0, which will be officially released within the next couple of days.

That means upstream autoscaler can only work with version >= 1.11.0 then?

**Comment by DmitriGekhtman** [comment]
> > Ray versions are updated to Ray 1.11.0, which will be officially released within the next couple of days.
> 
> That means upstream autoscaler can only work with version >= 1.11.0 then?

Correct.
I think when we declare this integration stable should in fact officially support only Ray >= 1.12.0 since the integration was put in place after the Ray 1.11.0 branch cut.

It would be feasible to loosen the version restriction if necessary, but some recent changes to the interface between the autoscaler and Ray make it a bit awkward.

**Comment by DmitriGekhtman** [comment]
The official Ray 1.11.0 release should be tomorrow, approximately.

**Comment by DmitriGekhtman** [comment]
Until the Ray 1.12.0 release, the autoscaler will use ray image built from this PR's merge commit.

**Comment by DmitriGekhtman** [comment]
After the Ray 1.12.0 release, the autoscaler should use a `rayproject/ray` image with the same Ray version as the cluster.

**Comment by DmitriGekhtman** [comment]
Update: 
Ray 1.11.0 has just been released!
https://github.com/ray-project/ray/releases

**Review by Jeffwan — APPROVED** [review]

**Comment by DmitriGekhtman** [comment]
Windows test_metrics is known to be flakey, unrelated.

**Comment by DmitriGekhtman** [comment]
rest of windows failures unrelated

**Comment by DmitriGekhtman** [comment]
Test failures unrelated.
Mac tests appear to be stalled -- these have already passed on a previous commit which differ from the current one by trivial changes to the new test file. 
Merging.

**Pull request merged**